===================================================================
RCS file: /home/eclipse/org.eclipse.swt/Eclipse SWT PI/cairo/library/cairo.c,v
retrieving revision 1.4
retrieving revision 1.5
diff -u -r1.4 -r1.5
--- cairo.c	2005/07/11 21:28:17	1.4
+++ cairo.c	2005/08/15 20:22:55	1.5
@@ -254,6 +254,62 @@
 }
 #endif
 
+#ifndef NO_cairo_1font_1options_1create
+JNIEXPORT jint JNICALL Cairo_NATIVE(cairo_1font_1options_1create)
+	(JNIEnv *env, jclass that)
+{
+	jint rc = 0;
+	Cairo_NATIVE_ENTER(env, that, cairo_1font_1options_1create_FUNC);
+	rc = (jint)cairo_font_options_create();
+	Cairo_NATIVE_EXIT(env, that, cairo_1font_1options_1create_FUNC);
+	return rc;
+}
+#endif
+
+#ifndef NO_cairo_1font_1options_1destroy
+JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1font_1options_1destroy)
+	(JNIEnv *env, jclass that, jint arg0)
+{
+	Cairo_NATIVE_ENTER(env, that, cairo_1font_1options_1destroy_FUNC);
+	cairo_font_options_destroy((cairo_font_options_t *)arg0);
+	Cairo_NATIVE_EXIT(env, that, cairo_1font_1options_1destroy_FUNC);
+}
+#endif
+
+#ifndef NO_cairo_1font_1options_1get_1antialias
+JNIEXPORT jint JNICALL Cairo_NATIVE(cairo_1font_1options_1get_1antialias)
+	(JNIEnv *env, jclass that, jint arg0)
+{
+	jint rc = 0;
+	Cairo_NATIVE_ENTER(env, that, cairo_1font_1options_1get_1antialias_FUNC);
+	rc = (jint)cairo_font_options_get_antialias((cairo_font_options_t *)arg0);
+	Cairo_NATIVE_EXIT(env, that, cairo_1font_1options_1get_1antialias_FUNC);
+	return rc;
+}
+#endif
+
+#ifndef NO_cairo_1font_1options_1set_1antialias
+JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1font_1options_1set_1antialias)
+	(JNIEnv *env, jclass that, jint arg0, jint arg1)
+{
+	Cairo_NATIVE_ENTER(env, that, cairo_1font_1options_1set_1antialias_FUNC);
+	cairo_font_options_set_antialias((cairo_font_options_t *)arg0, arg1);
+	Cairo_NATIVE_EXIT(env, that, cairo_1font_1options_1set_1antialias_FUNC);
+}
+#endif
+
+#ifndef NO_cairo_1get_1antialias
+JNIEXPORT jint JNICALL Cairo_NATIVE(cairo_1get_1antialias)
+	(JNIEnv *env, jclass that, jint arg0)
+{
+	jint rc = 0;
+	Cairo_NATIVE_ENTER(env, that, cairo_1get_1antialias_FUNC);
+	rc = (jint)cairo_get_antialias((cairo_t *)arg0);
+	Cairo_NATIVE_EXIT(env, that, cairo_1get_1antialias_FUNC);
+	return rc;
+}
+#endif
+
 #ifndef NO_cairo_1get_1current_1point
 JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1get_1current_1point)
 	(JNIEnv *env, jclass that, jint arg0, jdoubleArray arg1, jdoubleArray arg2)
@@ -309,6 +365,16 @@
 }
 #endif
 
+#ifndef NO_cairo_1get_1font_1options
+JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1get_1font_1options)
+	(JNIEnv *env, jclass that, jint arg0, jint arg1)
+{
+	Cairo_NATIVE_ENTER(env, that, cairo_1get_1font_1options_FUNC);
+	cairo_get_font_options((cairo_t *)arg0, (cairo_font_options_t *)arg1);
+	Cairo_NATIVE_EXIT(env, that, cairo_1get_1font_1options_FUNC);
+}
+#endif
+
 #ifndef NO_cairo_1get_1line_1cap
 JNIEXPORT jint JNICALL Cairo_NATIVE(cairo_1get_1line_1cap)
 	(JNIEnv *env, jclass that, jint arg0)
@@ -972,12 +1038,14 @@
 #endif
 
 #ifndef NO_cairo_1reference
-JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1reference)
+JNIEXPORT jint JNICALL Cairo_NATIVE(cairo_1reference)
 	(JNIEnv *env, jclass that, jint arg0)
 {
+	jint rc = 0;
 	Cairo_NATIVE_ENTER(env, that, cairo_1reference_FUNC);
-	cairo_reference((cairo_t *)arg0);
+	rc = (jint)cairo_reference((cairo_t *)arg0);
 	Cairo_NATIVE_EXIT(env, that, cairo_1reference_FUNC);
+	return rc;
 }
 #endif
 
@@ -1075,6 +1143,16 @@
 }
 #endif
 
+#ifndef NO_cairo_1set_1antialias
+JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1set_1antialias)
+	(JNIEnv *env, jclass that, jint arg0, jint arg1)
+{
+	Cairo_NATIVE_ENTER(env, that, cairo_1set_1antialias_FUNC);
+	cairo_set_antialias((cairo_t *)arg0, arg1);
+	Cairo_NATIVE_EXIT(env, that, cairo_1set_1antialias_FUNC);
+}
+#endif
+
 #ifndef NO_cairo_1set_1dash
 JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1set_1dash)
 	(JNIEnv *env, jclass that, jint arg0, jdoubleArray arg1, jint arg2, jdouble arg3)
@@ -1123,6 +1201,16 @@
 }
 #endif
 
+#ifndef NO_cairo_1set_1font_1options
+JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1set_1font_1options)
+	(JNIEnv *env, jclass that, jint arg0, jint arg1)
+{
+	Cairo_NATIVE_ENTER(env, that, cairo_1set_1font_1options_FUNC);
+	cairo_set_font_options((cairo_t *)arg0, (cairo_font_options_t *)arg1);
+	Cairo_NATIVE_EXIT(env, that, cairo_1set_1font_1options_FUNC);
+}
+#endif
+
 #ifndef NO_cairo_1set_1font_1size
 JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1set_1font_1size)
 	(JNIEnv *env, jclass that, jint arg0, jdouble arg1)
@@ -1371,14 +1459,12 @@
 #endif
 
 #ifndef NO_cairo_1surface_1finish
-JNIEXPORT jint JNICALL Cairo_NATIVE(cairo_1surface_1finish)
+JNIEXPORT void JNICALL Cairo_NATIVE(cairo_1surface_1finish)
 	(JNIEnv *env, jclass that, jint arg0)
 {
-	jint rc = 0;
 	Cairo_NATIVE_ENTER(env, that, cairo_1surface_1finish_FUNC);
-	rc = (jint)cairo_surface_finish((cairo_surface_t *)arg0);
+	cairo_surface_finish((cairo_surface_t *)arg0);
 	Cairo_NATIVE_EXIT(env, that, cairo_1surface_1finish_FUNC);
-	return rc;
 }
 #endif
 
@@ -1526,11 +1612,11 @@
 
 #ifndef NO_cairo_1xlib_1surface_1create_1for_1bitmap
 JNIEXPORT jint JNICALL Cairo_NATIVE(cairo_1xlib_1surface_1create_1for_1bitmap)
-	(JNIEnv *env, jclass that, jint arg0, jint arg1, jint arg2, jint arg3)
+	(JNIEnv *env, jclass that, jint arg0, jint arg1, jint arg2, jint arg3, jint arg4)
 {
 	jint rc = 0;
 	Cairo_NATIVE_ENTER(env, that, cairo_1xlib_1surface_1create_1for_1bitmap_FUNC);
-	rc = (jint)cairo_xlib_surface_create_for_bitmap((Display *)arg0, (Pixmap)arg1, arg2, arg3);
+	rc = (jint)cairo_xlib_surface_create_for_bitmap((Display *)arg0, (Pixmap)arg1, (Screen *)arg2, arg3, arg4);
 	Cairo_NATIVE_EXIT(env, that, cairo_1xlib_1surface_1create_1for_1bitmap_FUNC);
 	return rc;
 }
